Add support for reading v2 row level deletes in Iceberg connector (#21189)
prestodb/presto: presto-docs/src/main/sphinx/connector/iceberg.rst

    -----------+------+-----------+---------
    (0 rows)

> Table with deleted files

Reviewer: > deleted files

This sounds like files have been deleted... so it is strange that they can be read. I think you want to refer to Delete files, not deleted files. 

CC: @steveburnett

Reviewer 2: Thanks for this, @mbasmanova! I have reread the spec, the draft, and your comment, and I will follow up with a new review based on this. I appreciate your helping me understand this better.

> Table with deleted files

Reviewer 2: ```suggestion
Table with delete files
```

-----------------------

Iceberg V2 tables support row-level deletion. For more information see
`Row-level deletes <https://iceberg.apache.org/spec/#row-level-deletes>`_ in the Iceberg Table Spec.

> Presto supports reading deleted files, including Position Delete Files and Equality Delete Files.

Reviewer 2: ```suggestion
Presto supports reading delete files, including Position Delete Files and Equality Delete Files.
```

> When reading, Presto merges these deleted files to read the latest results.

Reviewer 2: ```suggestion
When reading, Presto merges these delete files to read the latest results.
```

    -----------+------+-----------+---------
    (0 rows)

> Table with delete files

Reviewer 2: ```suggestion
Table with deleted files
```

-----------------------

> Iceberg V2 tables support `row-level deletion <https://iceberg.apache.org/spec/#delete-formats>`,

Reviewer 2: ```suggestion
Iceberg V2 tables support row-level deletion. For more information see `Row-level deletes <https://iceberg.apache.org/spec/#row-level-deletes>`_ in the Iceberg Table Spec.  
```

Links can be tricky. This link did not build locally (see screenshot). 
<img width="889" alt="Screenshot 2023-10-31 at 10 48 50 AM" src="https://github.com/prestodb/presto/assets/7013443/fd9c65ed-6971-4b61-8aa9-b6f3547d3319">

Suggestion here edits the link for a successful local build, and revises the link introduction text following the [GitLab docs style guide for links](https://docs.gitlab.com/ee/development/documentation/styleguide/#links).

> and Presto supports reading these deleted files, including Position Delete Files and Equality Delete Files.

Reviewer 2: ```suggestion
Presto supports reading deleted files, including Position Delete Files and Equality Delete Files.
```
Suggest making this a separate sentence from the link in previous line.

> When reading, Presto will merge these deleted files to read the latest results.

Reviewer 2: ```suggestion
When reading, Presto merges these deleted files to read the latest results.
```
Revising sentence to present tense. For references, see the Google developer documentation style guide about [present tense](https://developers.google.com/style/tense).
